Sources/SwiftSyntax/SwiftSyntax.swift

```diff
@@ -22,56 +22,85 @@ import Glibc
 
 /// A list of possible errors that could be encountered while parsing a
 /// Syntax tree.
-public enum ParserError: Error {
+public enum ParserError: Error, CustomStringConvertible {
   case swiftcFailed(Int, String)
   case invalidFile
-}
 
-extension Syntax {
-  fileprivate static func encodeSourceFileSyntaxInternal(_ url: URL) throws -> Data {
-    let swiftcRunner = try SwiftcRunner(sourceFile: url)
-    let result = try swiftcRunner.invoke()
-    return result.stdoutData
+  public var description: String {
+    switch self{
```

Comment:
Done. (Any style errors in these commits are also in master, since I did straight cherrypicks without modification except in rare cases, but I'm happy to fix them up here.)